routes/search.js

```diff
@@ -0,0 +1,27 @@
+let bcrypt = require('bcrypt')
+let User = require('../models/User')
+let jwt = require('jsonwebtoken')
+
+module.exports = function (express) {
+  let router = express.Router()
+  
+  router.route('/') 
+    // requires user. Adds post to req.user
+    .get((req, res, next) => {
+      // check auth
+      if (!req.user) { return res.status(401).json({ error: 'Unauthorized' })}
+      console.log(`[garden] GET /api/search:`, req.query)
+      let query = {}
+      let searchTerm = req.query.displayName || req.query.username || ''
```

Comment:
Obviously that may be a bit overkill for a vast majority of cases. `username` is probably the only thing that's going to be searched for